### ISSUE TITLE
@include on inline Fragments generates non-nullable variable

### ISSUE THREAD

**Issue by bhavik-lfc** [apollographql/apollo-android#2796]
There are two variables quantity and purchaseValue which should be nullable since we have added include if.

ProductItemFragment.graphql 

```graphql
fragment ProductItem on Product {
//
... on Product @include(if: $isPurchased) {
    quantity
    purchaseValue
  }
}
```

but the generated kotlin classes has non-nullable variable for purchaseValue

```kotlin
data class ProductItem(
..
    val quantity: Int?,
    val purchaseValue: Int
)
```

Is there any way to make it nullable ?

In schema.json purchaseValue is defined as non-null

```json
...
{
              "name": "purchaseValue",
              "description": "",
              "args": [],
              "type": {
                "kind": "NON_NULL",
                "name": null,
                "ofType": {
                  "kind": "SCALAR",
                  "name": "Int",
                  "ofType": null
                }
              },
              "isDeprecated": false,
              "deprecationReason": null
            },
...
```

**Comment by martinbonnin** [comment]
One workaround would be to add the `@include` directives directly:

```
fragment ProductItem on Product {
  ... on Product {
    quantity @include(if: $isPurchased) 
    purchaseValue @include(if: $isPurchased) 
  }
}
```

That's more duplication in the GraphQL query but handling `@include` directive on fragments turns out to be pretty complicated. There can be multiple fragments on the same type condition with different `@include` directives meaning that we'd have to merge conditions somehow. It's doable but it'll be hard to do before 3.0

**Comment by bhavik-lfc** [comment]
👍  It works by adding `@include` directives.

But adding it on every variable would be difficult. Is there any plan for this after version 3.0 ?  :)

**Comment by martinbonnin** [comment]
Yep, I'm looking at this right now but 3.0 will change a bunch of things so it'll take some time to stabilize

**Comment by martinbonnin** [comment]
Some of the work to better support `@include/@skip` on fragments: https://github.com/apollographql/apollo-android/pull/2792/commits/152b0c7d7c6a38c71ca56e1e6ebe00a052ef7e7a

This allows to represent arbitrary combinations of `@include` directives, including `or` and `and`. This still needs to be implemented incodegen.

**Comment by martinbonnin** [comment]
@bhavik-lfc `3.0.0-alpha01` is out and should address this: https://github.com/apollographql/apollo-android/releases/tag/v3.0.0-alpha01. 

Can you give it a try and let us know how that worked?

**Comment by martinbonnin** [comment]
Closing as fixed in `3.0.0-alpha01`, let us know if you want to reopen.
